src/types/capabilities.rs

```diff
@@ -43,6 +44,17 @@ pub struct Capabilities {
     pub(crate) capabilities: HashSet<Capability<'this>>,
 }
 
+impl Clone for Capabilities {
+    fn clone(&self) -> Self {
+        // Give _rx a name so it's not immediately dropped. Otherwise any unsolicited responses
+        // that would be send there will return a SendError instead of the parsed response simply
+        // being dropped later.
```

Comment:
Let's also add a note here that any unsolicited items in here can safely be dropped because they must already have been communicated the first time the capabilities were parsed.